eventbus_test.go

```diff
@@ -196,9 +245,76 @@ func getConnectionCount(t *testing.T) int {
 
 	request.SetBasicAuth("guest", "guest")
 
-	restClient := http.Client{}
+	var apiResp rqmAPIResponse
 
-	resp, err := restClient.Do(request)
+	do := func() bool {
+		return len(apiResp) == expected
+	}
+
+	return compare(t, request, &apiResp, do)
+}
+
+func expectQueueConsumerCount(t *testing.T, queueName string, expected int) error {
+	t.Helper()
+
+	type queue struct {
+		Name      string `json:"name"`
+		Consumers int    `json:"consumers"`
+	}
+
+	type rqmAPIResponse []queue
+
+	request, err := http.NewRequest(http.MethodGet, "http://localhost:15672/api/queues", nil)
+	if err != nil {
+		t.Fatal("there should be no error:", err)
+	}
+
+	request.SetBasicAuth("guest", "guest")
+
+	var apiResp rqmAPIResponse
+
+	do := func() bool {
+		for i := range apiResp {
+			return apiResp[i].Name == queueName && apiResp[i].Consumers == expected
+		}
+
+		return false
+	}
+
+	return compare(t, request, &apiResp, do)
+}
+
+func compare(t *testing.T, request *http.Request, result any, compare func() bool) error {
```

Comment:
The func that is passed as a param and the func that receives the param are named the same, which could be confusing.